src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+
+## Inherit and override schema annotations
+But Rest.Li users found it useful to process annotations during schema processing. One use case is to introduce "inheritance" and "overrides" to annotations so those annotations can be dynamically processed in the way user defines, when for example the schemas were reused. This gives annotation extensibility and adds to schema reusability.
+
+Here are examples:
+Example case 1: Users might want the annotation of a field be inherited. The fields can reuse the annotations from upper level.
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, we might find both `visitedUrls` and `visitedUserProfiles` have `persistencePolicyInDays` annotated as 30.
+//<inheritance example>
+
+Example case 2: Users might want the annotation of a field be overriden. Override is where the annotation on a field or a schema might get updated when other annotations assign it another value.
+//<override example>
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, a user might find the field `visitedUserProfiles`s `persistencePolicyInDays` annotation is 30 but `visitedUrls`'s `persistencePolicyInDays` annotation has value 10, which overrides the original value inherited(which was 30).
+
+
+Example case 3:  Override might also happen when some annotation needs to be updated by another value assigned from another annotation location
+```pdl
+record UserVisitRecord {
+  //...
+  @persistencePolicyInDays = 365
+  recycledChatHistories: array[chat]
+
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  //...
+}
+
+record EnterpriseUserRecord {
+  userName: UserName
+  //...
+  @persistencePolicyInDays = {"/recycledChatHistories" : 3650}
+  visitRecord: UserVisitRecord
+  //...
+}
+```
+In this example, schema `EnterpriseUserRecord` reused `UserVisitRecord` and its annotation in the field `visitRecord`, but it overrides the annotation value of `recycledChatHistories`.
+
+
+All above examples shows that inheritance and overrides gives more extensibility to the schema annotations. Users should be free to define their own rules regarding how those annotations can be read. There should be an imambigous annotation value for the fields after user's rules are applied. We call this processing step to figure out the eventual value of a field or schema's annotation "resolution" and we call such value "resolved" value.
```

Comment:
imambigous -> unambiguous 

Also use third person:
We call this processing step to figure out the eventual value of a field or schema's annotation "resolution" and we call such value "resolved" value.

This processing step .. is called "resolution" and the final value is "resolved" value.